hawkular-javaagent/src/main/java/org/hawkular/agent/javaagent/JavaAgentEngine.java

```diff
@@ -235,7 +235,14 @@ public void start() {
 
     @Override
     public void stop() {
-        stopHawkularAgent();
+        // Ensure the JMX stop command returns successfully by doing this in a thread.
+        Runnable r = new Runnable() {
+            @Override
+            public void run() {
+                stopHawkularAgent();
+            }
+        };
+        new Thread(r).start();
```

Comment:
You know, let me try this without spawning the thread, that may be an unnecessary holdover from earlier approaches.  I think the main bit now is just not waiting for a response.  I'll test and let you know...